Makefile

```diff
@@ -23,19 +23,22 @@ depends3:
 
 .PHONY: pipinstallpythonapt
 pipinstallpythonapt: pipinstallpythonapt_deps
```

Comment:
[optional]

It took me a minute to realize that the pipinstallpythonapt* targets are only used for Travis, where "pip" means the pip in a virtualenv targeting the right Python version.  While it can wait, consider pulling these targets out into a separate, Travis-specific make file.

It may also make sense to be explicit about the Python version in the various calls to pip:

    $(PYTHON) -m pip install ...

This would require that we set PYTHON=... when running make in .travis.yml, as we are in the script at the bottom of that file.